In [1]:
#from __future__ import absolute_import, division, print_function, unicode_literals
import tensorflow as tf

In [2]:
sess = tf.Session(config=tf.ConfigProto(log_device_placement=True))
print(sess)

In [3]:
hello = tf.constant('Hello, TensorFlow!')

In [4]:
sess = tf.Session()
print(sess.run(hello))

b'Hello, TensorFlow!'


In [5]:
with tf.Session() as sess:
  devices = sess.list_devices()

In [6]:
print(devices)

[_DeviceAttributes(/job:localhost/replica:0/task:0/device:CPU:0, CPU, 268435456, 13190478191172464192), _DeviceAttributes(/job:localhost/replica:0/task:0/device:XLA_CPU:0, XLA_CPU, 17179869184, 13187599830489884004), _DeviceAttributes(/job:localhost/replica:0/task:0/device:GPU:0, GPU, 5911478272, 12951946658411127122), _DeviceAttributes(/job:localhost/replica:0/task:0/device:XLA_GPU:0, XLA_GPU, 17179869184, 10117507467157627103)]


[_DeviceAttributes(/job:localhost/replica:0/task:0/device:CPU:0, CPU, 268435456, 11047951367979657504), _DeviceAttributes(/job:localhost/replica:0/task:0/device:XLA_CPU:0, XLA_CPU, 17179869184, 9212443588184213143), _DeviceAttributes(/job:localhost/replica:0/task:0/device:GPU:0, GPU, 5911478272, 15457247141790241706), _DeviceAttributes(/job:localhost/replica:0/task:0/device:XLA_GPU:0, XLA_GPU, 17179869184, 13520540923942850894)]

In [7]:
print("GPU Available: ", tf.test.is_gpu_available())

GPU Available:  True


In [8]:
with tf.Session(config=tf.ConfigProto(log_device_placement=True))  as sess:
    devices = sess.list_devices()
print(devices)

[_DeviceAttributes(/job:localhost/replica:0/task:0/device:CPU:0, CPU, 268435456, 15650409004565514010), _DeviceAttributes(/job:localhost/replica:0/task:0/device:XLA_CPU:0, XLA_CPU, 17179869184, 14522417732166080971), _DeviceAttributes(/job:localhost/replica:0/task:0/device:GPU:0, GPU, 5911478272, 9820154598019495754), _DeviceAttributes(/job:localhost/replica:0/task:0/device:XLA_GPU:0, XLA_GPU, 17179869184, 5009890062130664103)]
